docs/notebooks/mappings.ipynb

```diff
@@ -0,0 +1,351 @@
+    "    print(\"Datasets hold the same values!\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's look more closely at the ICCLIM module\n",
+    "from xclim import icclim\n",
+    "import inspect\n",
+    "\n",
+    "print(icclim.__doc__)\n",
+    "for i in inspect.getmembers(icclim):\n",
+    "    if i[0][0] != '_':\n",
+    "        print(f\"{i[0]} : {i[1]}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating your own custom mapped module\n",
+    "\n",
+    "As we can see from the layout of ICCLIM, the module is effectively several wrapped instances of existing indices, with a majority comprising unchanged indices and several composed of indices with different threshold values.\n",
+    "\n",
+    "These mappings can be found in `xclim/xclim/__init__.py` and can be create using the `build_module` function. A mapoping requires two types of information:\n",
+    "\n",
+    "* A dictionary of mapped indices\n",
+    "* A docstring, usually written in ReStructuredText format\n",
+    "\n",
+    "Let's try creating our own custom indices for our specific purposes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from xclim.indicators import build_module\n",
+    "from xclim.core.utils import wrapped_partial\n",
+    "\n",
+    "def awesome_module(mode: str = \"ignore\"):\n",
+    "    mapping = dict(\n",
+    "        egg_cooking_season=wrapped_partial(indices.maximum_consecutive_tx_days, thresh=\"35 degC\"),\n",
+    "        precipitation_above_14mm=wrapped_partial(indices.wetdays, thresh=\"14.0 mm/day\"),\n",
+    "        sweater_weather=indices.tg_min\n",
+    "    )\n",
+    "    mod = build_module(\n",
+    "        \"xclim.awesome\",\n",
+    "        mapping,\n",
+    "        doc=\"\"\"\n",
+    "            =========================\n",
+    "            My Awesome Custom indices\n",
+    "            =========================\n",
+    "            There are only 3 indices that really matter when you come down to brass tacks.\n",
+    "            This mapping library exposes them to users who want to perform real deal \n",
+    "            climate science. Note that none of the checks performed by the `xclim.utils.Indicator` \n",
+    "            class (like with `xclim.atmos` indicators) are performed in this module.\n",
+    "\n",
+    "            \"\"\",\n",
+    "        mode=mode,\n",
+    "    )\n",
+    "    return mod\n",
+    "\n",
+    "AWESOME = awesome_module()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's look at our new awesome module\n",
+    "from xclim import awesome\n",
+    "import inspect\n",
+    "\n",
+    "print(awesome.__doc__)\n",
```

Comment:
Could use `help` here. Note that in the help, the partials are presented as data instead of functions. This could be investigated in another issue. Possibly boltons.funcutils could provide some improvements.

Reply:
Open a ticket about this?

Reply:
I do notice that the `help` function also hides the thresholds set by the mappings as well. It makes things a bit less clear for users who want to know what the default settings for mappings like ICCLIM are.

Reply:
@huard Is it alright if I leave you to open a ticket about this? I'll open one for the new `run_length` tests.

Reply:
This is being explored in #392